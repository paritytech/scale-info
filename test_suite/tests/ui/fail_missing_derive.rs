use scale_info::TypeInfo;

enum PawType<Paw> {
    Big(Paw),
    Small(Paw),
}
#[derive(TypeInfo)]
struct Cat<Tail, Ear, Paw> {
    tail: Tail,
    ears: [Ear; 3],
    paws: PawType<Paw>,
}

fn assert_type_info<T: TypeInfo + 'static>() {}

fn main() {
    assert_type_info::<Cat<bool, u8, u16>>();
}
