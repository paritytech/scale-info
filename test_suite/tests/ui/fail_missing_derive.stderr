error[E0277]: the trait bound `PawType<u16>: TypeInfo` is not satisfied
  --> $DIR/fail_missing_derive.rs:17:5
   |
17 |     assert_type_info::<Cat<bool, u8, u16>>();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `TypeInfo` is not implemented for `PawType<u16>`
   |
note: required because of the requirements on the impl of `TypeInfo` for `Cat<bool, u8, u16>`
  --> $DIR/fail_missing_derive.rs:7:10
   |
7  | #[derive(TypeInfo)]
   |          ^^^^^^^^
8  | struct Cat<Tail, Ear, Paw> {
   |        ^^^^^^^^^^^^^^^^^^^
note: required by a bound in `assert_type_info`
  --> $DIR/fail_missing_derive.rs:14:24
   |
14 | fn assert_type_info<T: TypeInfo + 'static>() {}
   |                        ^^^^^^^^ required by this bound in `assert_type_info`
   = note: this error originates in the derive macro `TypeInfo` (in Nightly builds, run with -Z macro-backtrace for more info)
